src/format/mod.rs

```diff
@@ -719,6 +796,81 @@ impl<'a, I: Iterator<Item = B> + Clone, B: Borrow<Item<'a>>> DelayedFormat<I> {
 impl<'a, I: Iterator<Item = B> + Clone, B: Borrow<Item<'a>>> fmt::Display for DelayedFormat<I> {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         format(f, self.date.as_ref(), self.time.as_ref(), self.off.as_ref(), self.items.clone())
+            .map_err(|_| fmt::Error)
+    }
+}
+
+/// A *temporary* object which can be used as an argument to `format!` or others.
+/// This is normally constructed via `format` methods of each date and time type.
+#[cfg(all(feature = "locales", any(feature = "alloc", feature = "std", test)))]
+#[derive(Debug)]
+pub struct DelayedFormatLocalized<'a, I> {
```

Comment:
I believe that you don't need to create this struct, you can just create a `Option<local>` field in the existing DelayedFormat and default it to `None` in the existing constructors, adding the new constructors under the feature flag.

Reply:
done